Packages/com.unity.inputsystem/InputSystem/InputSettings.cs

```diff
@@ -518,6 +530,119 @@ public ReadOnlyArray<string> supportedDevices
+            set
+            {
+                if (m_DisableMouseMoveCompression == value)
+                    return;
+
+                m_DisableMouseMoveCompression = value;
+                OnChange();
+            }
+        }
+
+        private void HookDisableMouseCompressionListener()
+        {
+            InputSystem.onActionChange += DisableMouseCompressionListener;
+        }
+
+        private void UnhookDisableMouseCompressionListener()
+        {
+            InputSystem.onActionChange -= DisableMouseCompressionListener;
+        }
+
+        private void DisableMouseCompressionListener(object o, InputActionChange change)
+        {
+            if (disableMouseMoveCompression)
+                return;
+
+            if (change != InputActionChange.BoundControlsChanged)
+                return;
+
+            switch (o)
+            {
+                case InputAction action:
+                    DetectMultiDeviceMouseMoveCompositeBinding(action);
+                    break;
+
+                case InputActionMap actionMap:
+                    foreach (var action in actionMap.actions)
+                    {
+                        DetectMultiDeviceMouseMoveCompositeBinding(action);
+                    }
+                    break;
+
+                case InputActionAsset asset:
+                    foreach (var action in asset.actionMaps.SelectMany(map => map.actions))
+                    {
+                        DetectMultiDeviceMouseMoveCompositeBinding(action);
+                    }
+                    break;
+            }
+        }
+
+        private void DetectMultiDeviceMouseMoveCompositeBinding(InputAction action)
+        {
+            if (!action.controls.Any(c => c.device is FastMouse))
+                return;
+
+            for (var i = 0; i < action.bindings.Count;)
+            {
+                var binding = action.bindings[i];
+                if (!binding.isComposite)
+                {
+                    ++i;
+                    continue;
+                }
+
+                var mousePositionBindingFound = false;
+                var otherDeviceBindingFound = false;
+                while (++i < action.bindings.Count)
+                {
+                    binding = action.bindings[i];
+
+                    if (binding.isPartOfComposite == false)
+                        break;
+
+                    var bindingPath = binding.path;
+                    if (string.Equals(bindingPath, "<mouse>/position", StringComparison.OrdinalIgnoreCase))
+                        mousePositionBindingFound = true;
+                    else if (!bindingPath.StartsWith("<mouse>", StringComparison.OrdinalIgnoreCase))
+                        otherDeviceBindingFound = true;
+                }
+
+                if (mousePositionBindingFound && otherDeviceBindingFound)
+                {
+                    Debug.LogWarning($"Mouse move event compression will be disabled because the mouse position control " +
```

Comment:
I think this regresses UX too much, many games have camera character/controls with shortcuts, which quite often are done with mouse position composites. Plus this code unlikely to catch someone writing a custom merger/interaction.
If you _really_ want to go that route - remove the warning. There is no point to show warning if it will trigger in many game projects, nor the warning is actionable for game developers, they do as per the manual.

I would go for original PR proposal and propagate every every into `FastMouse` (by that point you could already just batch all of them) so `FastMouse` can decide for itself when to flush the accumulation.

PS. Doing that on `onActionChange` feels _very_ brittle. Are you sure this will work with all serialization/domain reload magic there?

PS2. You forgot mouse delta, mouse scroll, mouse scroll delta :)

Reply:
Thanks for the comments. Yeah, good point about the warning. 

For custom interactions (what's a custom merger?), I don't see what's missing. If the user adds an interaction to a composite control, it will still get called at the right times.

> I would go for original PR proposal and propagate every every into FastMouse (by that point you could already just batch all of them) so FastMouse can decide for itself when to flush the accumulation.

If we go that route though we're mostly back to the more complicated solution from before. onActionChange does seem to catch everything, since it will fire from enabling an asset, map, or singleton control and I don't know of any way to get actions to work without enabling them.

> You forgot mouse delta, mouse scroll, mouse scroll delta :)

I did! Thanks.

Reply:
> If we go that route though we're mostly back to the more complicated solution from before

Not necessarily, IIRC `FastMouse.OnStateEvent` is already called for every event, so you might be able to copy your `Flush`  code and do that if non-mouse/non-state event arrived. That way checks for bindings are not necessary.

Reply:
Though that being said I'm a bit worried that if someone is listening to events they will still get all of them. Mouse compression in editor does that on windows event basis, so it eliminates problems from the source.

Reply:
> > If we go that route though we're mostly back to the more complicated solution from before
> 
> Not necessarily, IIRC `FastMouse.OnStateEvent` is already called for every event, so you might be able to copy your `Flush` code and do that if non-mouse/non-state event arrived. That way checks for bindings are not necessary.

My impression was that each event is only processed by one device. We look up the device by the deviceId property in each event, so FastMouse only ever sees mouse events.

Reply:
> Though that being said I'm a bit worried that if someone is listening to events they will still get all of them. Mouse compression in editor does that on windows event basis, so it eliminates problems from the source.

Yeah, that's a good point. This PR doesn't do anything to improve things like listen for unpaired device activity which is something we already have performance issues around. So I suppose we can either try and optimize those places further, or go with the more complicated approach for this PR.